snappiershot/compare.py

```diff
@@ -43,8 +45,16 @@ def _compare(self, value: Any, expected: Any, *, operations: List[Callable]) ->
               Tracks the operations that need to be applied to self.value and self.expected
                 to obtain value and expected, respectively. Used for logging differences.
         """
-        # Check the types of both objects.
-        if type(value) != type(expected):
+        # Special check for units since it's possible the types won't match even if they are equal objects
+        if isinstance(expected, Unit) or isinstance(value, Unit):
+            return self._compare_units(value, expected, operations=operations)
+
+        # Check the types of both objects
```

Comment:
perhaps instead of having to check float types here, move the `if isinstance(value, float) and not self.exact:` before this so all relevant float conditions are checked first and then you can remove the float check in the `if type` statement.

Reply:
done [545de73](https://github.com/MORSECorp/snappiershot/pull/96/commits/545de737c696ddb8936a7ad28021113b15ff6bd0)